sympy/solvers/solveset.py

```diff
@@ -531,8 +531,7 @@ def _solve_abs(f, symbol, domain):
         return ConditionSet(symbol, Eq(f, 0), domain)
 
 
-
-def _solveset(f, symbol, domain, _check=False):
+def _solveset(f, symbol, domain, _was_domain=False, _check=False,):
```

Comment:
I'm strongly against adding additional arguments to solveset unless it is **really** necessary, the issue you are trying to solve doesn't suggest that.